tests/unittests/test_handler/test_handler_puppet.py

```diff
@@ -177,3 +267,142 @@ def test_handler_puppet_writes_csr_attributes_file(self, m_subp, m_auto):
               pp_uuid: ED803750-E3C7-44F5-BB08-41A04433FE2E
             """)
         self.assertEqual(expected, content)
+
+    def test_puppet_runs_puppet_if_requested(self, m_auto):
+        """Run puppet with default args if 'exec' is set to True."""
+        with mock.patch.object(
+                subp, 'subp',
```

Comment:
`autospec` for the mocks in these tests too?

Reply:
I changed to mock with decorators, so from what I understand `autospec` shouldn't be needed anymore for these tests, but let me know if I'm wrong.

Reply:
Correct